src/config.rs

```diff
@@ -1,17 +1,128 @@
 pub mod settings;
 pub mod storage;
 
+use std::cell::Cell;
 use crate::config::settings::{Setting, SettingInfo};
 use crate::types::Result;
```

Comment:
This is your problem, you import the "own" Result enum and not the one from the rust stdlib.